client/src/components/ConversationModal.js

```diff
@@ -0,0 +1,128 @@
+ *                      alumni[{name, imageURL}], seen[bool], dateCreated }
+ *        modalOpen
+ *        closeModal()
+ */ 
+export default class Conversation extends Component {
+    constructor(props){
+        super(props)
+        this.state={
+            conversation: null,
+            display: [],
+            message: ''
+        }
+        this.handleValueChange = this.handleValueChange.bind(this)
+        this.sendMessage = this.sendMessage.bind(this)
+    }
+    componentWillMount() {
+        this.createDisplay(this.props.conversation)
+        this.setState({conversation: this.props.conversation})
+    }
+
+    async sendMessage() {
+        let conversation = await makeCall(
+            {
+                id: this.state.conversation._id,
+                timezone: parseInt(this.props.userDetails.timeZone/100),
+                message: this.state.message
+            }, '/conversations/sendMessage/' + this.props.userDetails._id, 'patch')
+        this.createDisplay(conversation.conversation)
+        this.setState({
+            conversation: conversation.conversation,
+            message: ''
+        })
+    }
+
+    createDisplay(conversation) {
+        let display = []
+        for (let message of conversation.messages) {
+            display.unshift(this.displayMessage(message, conversation.alumni, conversation.messages[0]))
+        }
+        this.setState({display: display})
+    }
+
+    handleValueChange(e, {name, value}) {
+        e.preventDefault();
+        this.setState({
+            [name]: value
+        })
+    }
+
+    displayMessage(message, alumni) {
+        let senderIndex = alumni.findIndex(item => item._id === message.senderId[0]);
+        let formattedMessage = message.message.split('\n').map((message, i) => {
+            return(
+                <p style={{'wordWrap': 'break-word', 'hyphens': 'auto', 'width': '50vw'}} key={i}>
+                    {message}
+                </p>
+                )
+        })
+        return (
+            <>
+            <Divider/>
+            <Feed.Event>
+                <Feed.Label><Image src={alumni[senderIndex].imageURL}/></Feed.Label>
+                <Feed.Content>
+                    <Feed.Summary>{alumni[senderIndex].name} <Feed.Date>{message.dateString}</Feed.Date></Feed.Summary>
+                    {formattedMessage}
+                </Feed.Content>
+            </Feed.Event>
+            </>
+        )
+    }
+
+    render() {
+        let userIndex = this.state.conversation.alumni.findIndex(item => item._id === this.props.userDetails._id);
+        let recipientIndex = (userIndex + 1) % 2;
+        return (
+            <Modal 
+                open={this.props.modalOpen} 
+                closeIcon 
+                onClose={this.props.closeModal}
+            >
+                <Modal.Header>Conversation with {this.state.conversation.alumni[recipientIndex].name}</Modal.Header>
```

Comment:
This is the only thing preventing groupchats from behaving properly at the moment - The solution when we eventually implement groupchats is to add (and x others) with a semantic ui popup to show their names